fleetdm/fleet
# Remove username from UI

- Remove username from Account Settings Page, Setup, Login
- Fix all related integration tests
- Prewire all e2e tests that should pass with API update

TODO:
- [x] `api/v1/setup` for `client_setup` endpoint needs to be changed to use `name` instead of `username`
- [x] activateuser e2e test should pass after API fix
- [x] setup e2e test should pass after API fix
- [x] Modify e2e `commands.ts` to use name/email instead of username (`login`, `loginSSO`) to match new API
- [ ] Manual QA Setup
- [ ] Manual QA Login
- [ ] Manual QA `UserSettingPage`
- [x] Thinking of allowing the login page to still accept legacy usernames even if the UI only states Email
- [ ] Confirm mocks



Addresses #1002 
NEW SCREENSHOTS:
<img width="1354" alt="Screen Shot 2021-06-22 at 5 14 29 PM" src="https://user-images.githubusercontent.com/71795832/123016294-216c8700-d37f-11eb-8079-30b9bb397850.png">
<img width="1362" alt="Screen Shot 2021-06-22 at 5 05 44 PM" src="https://user-images.githubusercontent.com/71795832/123016303-23cee100-d37f-11eb-86c8-ebadf751dec1.png">
<img width="1357" alt="Screen Shot 2021-06-23 at 7 47 36 AM" src="https://user-images.githubusercontent.com/71795832/123118335-659b6e00-d3f7-11eb-9e61-12688b3be1c2.png">
6/23 - New user settings page matches figma update
6/22 - On another note, I rearranged the order flow for this page to de-emphasize username. Also, I reordered our setup form to be Full name, email, password, password since before it was username password password email, which felt like it needed reordering with the change



Running a manual QA on everything with console and server console open. Feel free to add more stuff to QA
<table>
<tr>
<th>Manual QA</th>
  <th>🐩<br>ZW</th>
  <th>🦘<br>RP</th>
 </tr>
<tr>
<td>

- Setup
- fleetctl setup
- Log in
- Log out
- Invite user
- Log in invite user
- Create sso user
- Login sso user
- Create and log in API Only user
- Create queries
- Create packs
- Create and assign users to teams (basic)
- Switch users on teams (basic)
- Edit global user (core)
- Edit team user (basic)
- Successfully edit current user name, email(SMTP must be configured for email change), password
- Seed core testing users
- Require user to password reset

</td>

<td>

- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y
- [x] y

</td>

<td>

- [x] y
- [ ] y
- [x] y
- [x] y
- [x] y
- [ ] y
- [ ] y
- [ ] y
- [ ] y
- [x] y
- [x] y
- [ ] y
- [ ] y
- [ ] y
- [x] y
- [x] y
- [x] y
- [x] y

</td>

</tr>
</table>

Bugs found:
- Still want to render an error message if someone tries to update their own email without SMTP configured.
